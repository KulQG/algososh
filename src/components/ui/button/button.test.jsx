import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from '@testing-library/react';
import { Button } from "./button";

describe('Button', () => {
  it("Кнопка рендерится без ошибок", () => {
    const tree = renderer
      .create(<Button />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопка рендерится без ошибок", () => {
    const tree = renderer
      .create(<Button text="Кнопка" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопка рендерится без ошибок", () => {
    const tree = renderer
      .create(<Button text="Кнопка" disabled />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопка рендерится без ошибок", () => {
    const tree = renderer
      .create(<Button isLoader />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Коллбэк кнопки работает как надо', () => {
    const onClickMock = jest.fn()

    const { getByText } = render(<Button text="clickMe" onClick={onClickMock} />)
    const button = getByText('clickMe')

    fireEvent.click(button)
    expect(onClickMock).toHaveBeenCalledTimes(1);
  })

  test('Коллбек кнопки работает как надо', () => {
    const onClickMock = jest.fn()

    const { getByText } = render(<Button text="clickMe" onClick={onClickMock} />)
    const button = getByText('clickMe')

    fireEvent.click(button)
  })

  test('Тест вызова алерта', () => {
    const alertMock = jest.fn();
    window.alert = alertMock;

    const { getByText } = render(
      <Button onClick={() => window.alert('Button clicked!')} text="Click me" />
    );

    const button = getByText('Click me');
    fireEvent.click(button);

    expect(alertMock).toHaveBeenCalledTimes(1);
    expect(alertMock).toHaveBeenCalledWith('Button clicked!');

    window.alert = globalThis.alert;
  });
})

