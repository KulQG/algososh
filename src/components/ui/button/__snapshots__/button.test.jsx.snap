// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Кнопка рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button Кнопка рендерится без ошибок 2`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка
  </p>
</button>
`;

exports[`Button Кнопка рендерится без ошибок 3`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка
  </p>
</button>
`;

exports[`Button Кнопка рендерится без ошибок 4`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
